app/models/protocol.rb

```diff
@@ -504,13 +504,15 @@ def has_line_items_of_type?(current_request, portal, type)
   end
 
   def direct_cost_total(service_request)
```

Comment:
Refactor SQL queries and added eager load for the `direct_cost_total` and `indirect_cost_total` methods